src/broker/messaging.bif

```diff
@@ -208,13 +204,10 @@ function Cluster::publish_rr%(pool: Pool, key: string, ...%): bool
 	if ( ! topic->AsString()->Len() )
 		return zeek::val_mgr->False();
 
-	const auto& bif_args = @ARGS@;
-	ValPList args(bif_args->size() - 2);
+	auto first = @ARGS@->begin();
+	std::advance(first, 2);
 
-	for ( auto i = 2u; i < bif_args->size(); ++i )
-		args.push_back((*bif_args)[i].get());
-
-	auto rval = publish_event_args(args, topic->AsString(), frame);
+	auto rval = publish_event_args(first, @ARGS@->end(), topic->AsString(), frame);
```

Comment:
This can be done in one shot:


```c++
auto rval = publish_event_args(std::next(@ARGS@->begin(), 2), @ARGS@->end(), topic->AsString(), frame);
```